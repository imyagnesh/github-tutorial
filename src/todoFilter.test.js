import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoFilter from "./todoFilter";

test("should take snapshot", () => {
  const { container } = render(<TodoFilter value="all" onFilter={() => {}} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("should render TodoFilter Without Error", () => {
  const { container } = render(<TodoFilter value="all" onFilter={() => {}} />);
  expect(container.firstChild).not.toBeNull();
});

test("should render 3 buttons", () => {
  const { queryAllByRole } = render(
    <TodoFilter value="all" onFilter={() => {}} />
  );
  const buttons = queryAllByRole("button");
  expect(buttons.length).toBe(3);
});

test("should click all button", () => {
  const fn = jest.fn();
  const { queryAllByRole } = render(<TodoFilter value="all" onFilter={fn} />);
  const buttons = queryAllByRole("button");
  fireEvent.click(buttons[0]);
  expect(fn).toBeCalledTimes(1);
  expect(fn).toBeCalledWith("all");
  fireEvent.click(buttons[1]);
  expect(fn).toBeCalledTimes(2);
  expect(fn).toBeCalledWith("pending");
  fireEvent.click(buttons[2]);
  expect(fn).toBeCalledTimes(3);
  expect(fn).toBeCalledWith("completed");
});
