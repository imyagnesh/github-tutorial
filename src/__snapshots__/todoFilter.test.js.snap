// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take snapshot 1`] = `
<div
  class="MuiTabs-root"
>
  <div
    class="MuiTabs-scroller MuiTabs-fixed"
    style="overflow: hidden;"
  >
    <div
      aria-label="icon label tabs example"
      class="MuiTabs-flexContainer"
      role="tablist"
    >
      <button
        aria-selected="true"
        class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected MuiTab-fullWidth"
        role="button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiTab-wrapper"
        >
          All Todos
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected="false"
        class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary MuiTab-fullWidth"
        role="button"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiTab-wrapper"
        >
          Pending
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected="false"
        class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary MuiTab-fullWidth"
        role="button"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiTab-wrapper"
        >
          Completed
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <span
      class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
      style="left: 0px; width: 0px;"
    />
  </div>
</div>
`;
